security_scheme.go

```diff
@@ -37,8 +37,8 @@ func BasicAuth() *SecurityScheme {
 }
 
 // APIKeyAuth creates an api key auth security scheme
-func APIKeyAuth(fieldName, valueSource string) *SecurityScheme {
-	return &SecurityScheme{SecuritySchemeProps: SecuritySchemeProps{Type: apiKey, Name: fieldName, In: valueSource}}
+func APIKeyAuth(fieldName, valueSource, fieldDescription string) *SecurityScheme {
+	return &SecurityScheme{SecuritySchemeProps: SecuritySchemeProps{Type: apiKey, Name: fieldName, In: valueSource, Description: fieldDescription}}
```

Comment:
`auth.SecuritySchemeProps.Description = "my description"`
Should be able to set the field without change the function.

Reply:
I agree. If it really needs to be in here, then I would prefer another method instead of breaking the interface.

You could add a `WithDescription` method that takes care of this for a fluent builder style api